packages/code-infra/src/cli/cmdBuild.mjs

```diff
@@ -69,16 +68,16 @@ ${content}`,
 
 /**
  * @param {Object} param0
- * @param {string | Record<string, string>} param0.importPath
+ * @param {NonNullable<import('./packageJson').PackageJson.Exports>} param0.importPath
  * @param {string} param0.key
  * @param {string} param0.cwd
  * @param {string} param0.dir
  * @param {string} param0.type
- * @param {Object} param0.newExports
+ * @param {import('./packageJson').PackageJson.ExportConditions} param0.newExports
  * @param {string} param0.typeOutExtension
  * @param {string} param0.outExtension
  * @param {boolean} param0.addTypes
- * @returns {Promise<{path: string[], importPath: string | Record<string, string | undefined>}>}
+ * @returns {Promise<void>}
```

Comment:
Some of the props were applied in the function, some outside, and some both. Let's just update `newExports` inside of the function to simplify the flow?